syntax = "proto3";

package  tv.uob.grpc;

option java_multiple_files = true;
option java_package = "tv.uob.grpc";

message User {
    int32 id = 1;
    string username = 2;
    string email = 3;
    string passwordHash = 4;
    Profile profile = 5;
}

message Profile {
    string avatarURL = 1;
}

message GetUserByIdRequest {
    int32 id = 1;
}

message GetUserByEmailRequest {
    string email = 1;
}

message GetUserByUsernameRequest {
    string username = 1;
}

message CreateUserRequest {
    string username = 1;
    string email = 2;
    string passwordHash = 3;
}

message CreateUserResponse {
    bool created = 1;
}


service UserService {
    rpc GetById(GetUserByIdRequest) returns (User) {}
    rpc GetByUsername(GetUserByUsernameRequest) returns (User) {}
    rpc GetByEmail(GetUserByEmailRequest) returns (User) {}
    rpc CreateUser(CreateUserRequest) returns (CreateUserResponse) {}
}
